File: 2239_find_closest_number_to_zero/2239_find_closest_number_to_zero.c
int findClosestNumber(int* nums, int numsSize) {
    int retVal = nums[0];

    for(int i = 1; i<numsSize; i++) {
        if(abs(retVal) > abs(nums[i]))
        {
            retVal = nums[i];
        }
        else if((abs(retVal) == abs(nums[i])) && (nums[i] > retVal))
        {
            retVal = nums[i];
        }
    }

    return retVal;
}

File: 2239_find_closest_number_to_zero/2239_find_closest_number_to_zero.py
class Solution:
    def findClosestNumber(self, nums: List[int]) -> int:
        closest = float('inf')
        for x in nums:
            if abs(x) < abs(closest):
                closest = x
        
        if closest < 0 and abs(closest) in nums:
            return abs(closest)
        return closest